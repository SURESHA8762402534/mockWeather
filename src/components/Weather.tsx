import { Card, Typography } from '@mui/material'
import axios from 'axios'
import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'

const Weather = () => {
    const [data, setData]:any = useState()
    const {capital} = useParams()

    const fetchWeather = async()=>{
        try{
            const res = await axios.get(`http://api.weatherstack.com/current?access_key=d81381e7f65ac20a3115570356e0a315&query=${capital}`)
       console.log(res.data.current);
       setData(res.data.current)
        }
        catch(e){
            console.log(e);
        }
    }

useEffect(()=>{
    fetchWeather()
},[capital])

  return (
    
    <>
    <h3 style={{textAlign:'center'}}>Weather Details of {capital}</h3>
    {data ? 
    <Card sx={{p:4,ml:60, mt:4}} style={{width:'30vw', boxShadow:'5px 10px #888888'}}>
    <div style={{float:'left'}}>
        <Typography variant='h5'>
            Capital : {capital}
        </Typography>
        <Typography variant='h5'>
            Temparature : {data.temperature}<sup>0</sup> Celcius
        </Typography>
        <Typography variant='h5'>
            Wind Speed : {data.wind_speed} km/hr
        </Typography>
        <Typography variant='h5'>
            Precip : {data.precip}
        </Typography>
    </div>
    <div style={{float:'right'}}>
        <img src={data.weather_icons[0]}/>
    </div>
</Card>
:
null
}
    </>
  )
}

export default Weather