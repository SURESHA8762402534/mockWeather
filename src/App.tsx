/** @format */
import "./App.css";
import { Route, Routes, useNavigate } from "react-router-dom";
import axios from "axios";
import Nav from "./components/Nav";
import Country from "./components/Country";
import Weather from "./components/Weather";

function App() {

	const navigate = useNavigate()

	return (
	<>
	<Nav navigate={navigate}/>
	<Routes>
		<Route path="/country/:country" element={<Country/>}/>
		<Route path='/weather/:capital' element={<Weather/>}/>
	</Routes>
	</>
	);
}

export default App;