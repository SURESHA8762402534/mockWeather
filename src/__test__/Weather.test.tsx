import { screen, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../App";
import Weather from "../components/Weather";

it('snap for weather',()=>{
    const tree = render(
        <BrowserRouter>
        <Weather/>
        </BrowserRouter>
    );
    expect(tree).toMatchSnapshot()
})

it('weather component test',()=>{
    render(
        <BrowserRouter>
        <Weather/>
        </BrowserRouter>
    );
   screen.debug()
})

test('for app component',()=>{
    const tree = render(
        <BrowserRouter>
       <App/>
        </BrowserRouter>
    );

    screen.debug()
    expect(tree).toMatchSnapshot()
})