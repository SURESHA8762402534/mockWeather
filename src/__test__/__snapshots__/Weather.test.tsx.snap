// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for app component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1vg4bp-MuiTypography-root"
          >
            Weather Report
          </div>
          <form
            action=""
          >
            <label
              for="input"
            >
              Country 
            </label>
            <input
              id="input"
              name="name"
              placeholder="enter country"
              style="height: 2rem;"
              type="text"
            />
            <button
              style="height: 2.4rem;"
              type="submit"
            >
              Search
            </button>
          </form>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1vg4bp-MuiTypography-root"
        >
          Weather Report
        </div>
        <form
          action=""
        >
          <label
            for="input"
          >
            Country 
          </label>
          <input
            id="input"
            name="name"
            placeholder="enter country"
            style="height: 2rem;"
            type="text"
          />
          <button
            style="height: 2.4rem;"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snap for weather 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        style="text-align: center;"
      >
        Weather Details of 
      </h3>
    </div>
  </body>,
  "container": <div>
    <h3
      style="text-align: center;"
    >
      Weather Details of 
    </h3>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
